src/mpi/datatype/type_dup.c

```diff
@@ -62,12 +62,12 @@ int MPIR_Type_dup(MPI_Datatype oldtype, MPI_Datatype * newtype)
         new_dtp->typerep.handle = NULL;
         *newtype = new_dtp->handle;
 
+        mpi_errno = MPIR_Typerep_create_dup(oldtype, new_dtp);
+        MPIR_ERR_CHECK(mpi_errno);
+
         if (old_dtp->is_committed) {
             MPID_Type_commit_hook(new_dtp);
         }
-
-        mpi_errno = MPIR_Typerep_create_dup(oldtype, new_dtp);
-        MPIR_ERR_CHECK(mpi_errno);
```

Comment:
I believe this is only a dataloop weirdness.  Yaksa doesn't require this special commit after dup.

Reply:
Yeah.

Reply:
So, perhaps this can be moved to dataloop?